mitodl/mit-open
# shanbady/separate database router and schema for program certificates

### What are the relevant tickets?

Closes [#3771](https://github.com/mitodl/hq/issues/3771)



### Description (What does it do?)
The changes in this PR properly migrate the ProgramCertificate table to its own schema and also includes a database router that prevents writes from happening on tables in the external schema.

### How can this be tested?
1. checkout this branch
2. run the migrations or do a full docker-compose down+up
3. in the postgres database see that there is an external schema. in manage.py dbshell you can run the following:
```
 SELECT nspname
 FROM pg_catalog.pg_namespace;
```
4. trying to create a programcertificate in django shell manually should give you an exception:
```
from profiles.models import ProgramCertificate
ProgramCertificate.objects.get_or_create(user_full_name='john doe',user_email='test@test.com', micromasters_program_id=1, program_title='Supply Chain Management')
```

### Additional Context
Our previous deploy of the programcertificate table ended up create a table in the public schema named "external.programcertificate".

The new migrations in this PR do the following to fix that:
1. [Drop the rogue programcertificate table](https://github.com/mitodl/mit-open/blob/shanbady/separate-database-router-for-program-certificates2/profiles/migrations/0018_alter_programletter_certificate_and_more.py#L12) 
2. [recreate the programcertificate table](https://github.com/mitodl/mit-open/blob/shanbady/separate-database-router-for-program-certificates2/profiles/migrations/0018_alter_programletter_certificate_and_more.py#L13) in the external schema
